### PR TITLE
update docs

### PR DESCRIPTION
**Pull request by pacman100**
### What does this PR do?
1. Addresses comment https://github.com/huggingface/accelerate/pull/667#discussion_r993500672
2. Updates training zoo to include AWS SageMaker and Megatron-LM examples.

### CODE REVIEW

**Review by muellerzr — APPROVED** [review]
Thanks!

File: docs/source/usage_guides/megatron_lm.mdx [hunk 1]
```diff
@@ -23,18 +23,45 @@ For detailed information and how things work behind the scene please refer the g
 Accelerate integrates following feature of Megatron-LM to enable large scale pre-training/finetuning
 of BERT (Encoder), GPT (Decoder) or T5 models (Encoder and Decoder):
 
-a. **Tensor Parallelism (TP)**: Reduces memory footprint without much additional communication on intra-node ranks
+a. **Tensor Parallelism (TP)**: Reduces memory footprint without much additional communication on intra-node ranks.
+Each tensor are split into multiple chunks with each shard residing on separate GPU. At each step, same minimbatch of data is processed
```
Comment (sgugger) [inline docs/source/usage_guides/megatron_lm.mdx:27]:
```suggestion
Each tensor is split into multiple chunks with each shard residing on separate GPU. At each step, the same mini-batch of data is processed
```

File: docs/source/usage_guides/megatron_lm.mdx [hunk 2]
```diff
@@ -23,18 +23,45 @@ For detailed information and how things work behind the scene please refer the g
...
+Layers are distributed uniformly across PP stages. For example, if a model has `24` layers and we have `4` GPUs for
+pipeline parallelism, each GPU will have `6` layers (24/4). For more details on schedules to reduce the idle time of PP,
+please refer to the research paper [Efficient Large-Scale Language Model Training on GPU Clusters
+Using Megatron-LM](https://arxiv.org/pdf/2104.04473.pdf) and 
+this section of 🤗 blogpost [The Technology Behind BLOOM Training](https://huggingface.co/blog/bloom-megatron-deepspeed#pipeline-parallelism).
 
 c. **Sequence Parallelism (SP)**: Reduces memory footprint without any additional communication. Only applicable when using TP.
-
-d. **Data Parallelism (DP)** via Distributed Optimizer: Reduces memory footprint by sharding optimizer states and gradients across DP ranks.
+It reduced activation memory required as it prevents same copies to be on the tensor parallel ranks
```
Comment (sgugger) [inline docs/source/usage_guides/megatron_lm.mdx:44]:
```suggestion
It reduces activation memory required as it prevents the same copies to be on the tensor parallel ranks 
```

File: docs/source/usage_guides/megatron_lm.mdx [hunk 3]
```diff
@@ -23,18 +23,45 @@ For detailed information and how things work behind the scene please refer the g
...
-d. **Data Parallelism (DP)** via Distributed Optimizer: Reduces memory footprint by sharding optimizer states and gradients across DP ranks.
+It reduced activation memory required as it prevents same copies to be on the tensor parallel ranks 
+post `all-reduce` by replacing it with `reduce-scatter` and `no-op` operation would be replaced by `all-gather`. 
+As `all-reduce = reduce-scatter + all-gather`, this saves a ton of activation memory at no added communication cost. 
+To put it simply, it shards the outputs of each transformer layer along sequence dimension, e.g., 
+if the sequence length is `1024` and the TP size is `4`, each GPU will have `256` tokens (1024/4) for each sample. 
+This increases the batch size that can be supported for training. For more details, please refer to the research paper
+[Reducing Activation Recomputation in Large Transformer Models](https://arxiv.org/pdf/2205.05198.pdf). 
+
+d. **Data Parallelism (DP)** via Distributed Optimizer: Reduces memory footprint by sharding optimizer states and gradients across DP ranks
```
Comment (sgugger) [inline docs/source/usage_guides/megatron_lm.mdx:52]:
```suggestion
d. **Data Parallelism (DP)** via Distributed Optimizer: Reduces the memory footprint by sharding optimizer states and gradients across DP ranks
```

**Review by sgugger — APPROVED** [review]
Thanks a lot! Left a few comments to fix typos.

**Pull request merged**